### PR TITLE
refs #3345 fixed replace()

### PR DESCRIPTION
**Pull request by davidnich**

### CODE REVIEW

File: lib/ql_string.qpp [hunk 1]
```diff
@@ -1205,43 +1214,55 @@ string replace(string str, string source, string target, int start = 0, int end
         const char* pattern = t1->c_str();
 
         if (start) {
+            if (start < 0) {
+                start = str->size() + start;
```
Comment (omusil24) [inline lib/ql_string.qpp]: The `start` and `end` params should be in characters (not bytes). `QoreString::size()` returns size in bytes though.

**Comment by omusil24** [comment]
Added some tests, the 2 last ones are failing.

File: lib/ql_string.qpp (hunk 1 above)
Comment (davidnich) [reply lib/ql_string.qpp]: fixed

**Comment by omusil24** [comment]
@davidnich I added a note to the replace function documentation + changed the return type to `*string`, since the function can return NOTHING. Is this okay?

**Comment by davidnich** [comment]
> @davidnich I added a note to the replace function documentation + changed the return type to `*string`, since the function can return NOTHING. Is this okay?

@omusil24 yes I see there is an inconsistency there, in fact I think this function should return the original string in that place so that the return value can stay `string`; I will fix that and push the changes now

**Pull request merged**